checkov/cdk/checks/python/APIGatewayAuthorization.yaml

```diff
@@ -0,0 +1,14 @@
+metadata:
+  version: 0.2
+  approach: define failing
+  id: CKV_AWS_59
+  name: Ensure there is no open access to back-end resources through API
+  category: GENERAL_SECURITY
+scope:
+  languages:
+    - python
+definition:
+  pattern: aws_cdk.aws_apigateway.Method(<ANY>, http_method=$ARG, <ANY>, authorization_type=aws_cdk.aws_apigateway.AuthorizationType.NONE, <ANY>, api_key_required=False)
```

Comment:
Keyword arguments can be supplied everywhere in the function call..
You need to create SIX patterns (with or between them).
http_method, authorization_type, api_key_required
http_method, api_key_required, authorization_type
api_key_required, http_method, authorization_type
api_key_required, authorization_type, http_method
 authorization_type, api_key_required, http_method
 authorization_type, http_method,  api_key_required
 
Send it to @tsmithv11